libs/design-system/src/components/ClientSideOnly.tsx

```diff
@@ -0,0 +1,18 @@
+'use client'
+
+import { useEffect, useState } from 'react'
+
+/**
+ * A wrapping component that forces its children to render on the client side.
+ */
+export function ClientSideOnly({ children }: { children: React.ReactNode }) {
+  const [shouldRender, setShouldRender] = useState(false)
+
+  useEffect(() => {
+    setShouldRender(true)
+  }, [])
+
+  if (!shouldRender) return null
```

Comment:
Relating to your other comment, `fallback` could be rendered here:
`if (!shouldRender) return fallback || null`